x-pack/solutions/security/plugins/security_solution/public/detections/components/common/search_bar/search_bar_section.tsx

```diff
@@ -26,7 +26,7 @@ export interface SearchBarSectionProps {
 }
 
 /**
- * UI section of the alerts page that renders the global search bar.
+ * UI section of the detections page (alerts and attacks) that renders the global search bar.
  */
 export const SearchBarSection = memo(
```

Comment:
Same for this file, while I see a small advantage in making this component reusable, I don't think it makes sense as part of this PR. Why using this new component on the alerts and attacks pages, but not in any of the other pages in Security Solution (like the entity analytics details, hosts, network, users, rule details...) which are also using the same code.

I think this should be done in a separate PR and touch a bigger scope.

Reply:
If you want to keep this anyway, I think we should add a new prop to be able to prefix the dataTestSubj 

```typescript
/**
 * Optional data test subject string
 */
'data-test-subj'?: string;

{...}

({ dataView, sourcererDataViewSpec, 'data-test-subj': dataTestSubj }: SearchBarSectionProps)

{...}

<SiemSearchBar
  dataTestSubj={`${dataTestSubj}-${SEARCH_BAR_TEST_ID}`}
```

That way we're guaranteed to have unique ids in Security Solution.